x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1945,14 +1945,19 @@ private void reloadSharedSecretsForJwtRealms(Settings settingsWithKeystore) {
      * See {@link TransportReloadRemoteClusterCredentialsAction} for more context.
      */
     private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
+        // Using `settings` instead of `settingsWithKeystore` is deliberate: we are not interested in secure settings here
+        if (DiscoveryNode.isStateless(settings)) {
+            // Stateless does not support remote cluster operations. Skip.
+            return;
+        }
+
         final PlainActionFuture<ActionResponse.Empty> future = new PlainActionFuture<>();
```

Comment:
Can we short circuit this entirely if the credentials did not change ?

Reply:
It's tricky: just the absence of credentials in the settings doesn't mean a noop (it could be that we removed credentials) -- anything else would require access to the remote cluster service which is not available here. I wanted to add a short-circuit here as well originally but didn't come up with a good way to do it.